admin/includes/javascript/orders.php

```diff
@@ -0,0 +1,42 @@
+<?php
+// Copy address details to clipboard
+if (isset($action) && $action === 'edit') { ?>
+<script>
+    function copyToClipboard(addressType) {
+        <?php
+        $address_types = ['customer', 'delivery', 'billing'];
+        foreach ($address_types as $address_type) {
+            $addressBlock[$address_type] =
+                $order->$address_type['name'] . "\\r\\n" .
+                (empty($order->$address_type['company']) ? '' : $order->$address_type['company'] . "\\r\\n") .
+                $order->$address_type['street_address'] . "\\r\\n" .
+                (empty($order->$address_type['suburb']) ? '' : $order->$address_type['suburb'] . "\\r\\n") .
+                (empty($order->$address_type['city']) ? '' : $order->$address_type['city'] . "\\r\\n") .
+                $order->$address_type['state'] . "\\r\\n" .
+                $order->$address_type['postcode'] . "\\r\\n" .
+                $order->$address_type['country']['title'] . "\\r\\n" .
+                (empty($order->$address_type['telephone']) ? '' : $order->$address_type['telephone']) . "\\r\\n" .
+                (empty($order->$address_type['email_address']) ? '' : $order->$address_type['email_address']);
+        }
+        ?>
+        let address;
+        switch (addressType) {
+            case('customer') :
+                address = '<?= $addressBlock['customer'] ?>';
+                break;
+            case('delivery') :
+                address = '<?= $addressBlock['delivery'] ?>';
+                break;
+            case('billing') :
+                address = '<?= $addressBlock['billing'] ?>';
+                break;
+            default:
+                return;
+        }
+        navigator.clipboard.writeText(address).then(r => '');
+        // show the copied text
+        alert('"' + addressType + '":' + "\n" + address);
```

Comment:
Is this alert intentionally showing a confirmation pop-up that the user must acknowledge?
Thinking about UX here.

Perhaps updating the button text to say "Copied!" might be sufficient acknowledgement?